Assets/Scripts/Model/Content/SecondEdition/Upgrades/Gunner/AirenCracken.cs

```diff
@@ -0,0 +1,159 @@
+﻿using Ship;
+using SubPhases;
+using System;
+using System.Collections.Generic;
+using Tokens;
+using Upgrade;
+
+namespace UpgradesList.SecondEdition
+{
+    public class AirenCracken : GenericUpgrade
+    {
+        public AirenCracken() : base()
+        {
+            IsHidden = true;
+
+            UpgradeInfo = new UpgradeCardInfo(
+                "Airen Cracken",
+                types: new List<UpgradeType>() { UpgradeType.Gunner },
+                cost: 0,
+                isLimited: true,
+                abilityType: typeof(Abilities.SecondEdition.AirenCrackenGunnerAbility)
+            );
+
+            NameCanonical = "airencracken-boe";
+            ImageUrl = "https://raw.githubusercontent.com/sampson-matt/FlyCasualLegacyCustomCards/refs/heads/main/BattleOverEndor/AirenCracken.jpg";
+        }
+    }
+}
+
+namespace Abilities.SecondEdition
+{
+    //After defending, if the attack hit, you may gain 1 deplete token to perform a bonus primary attack targeting the attacker.
+    public class AirenCrackenGunnerAbility : GenericAbility
+    {
+        private bool IsPerformedRegularAttack;
+        private GenericShip ShipToPunish;
+        public override void ActivateAbility()
+        {
+            HostShip.OnAttackHitAsDefender += PlanBonusAttack;
+        }
+
+        public override void DeactivateAbility()
+        {
+            HostShip.OnAttackHitAsDefender -= PlanBonusAttack;
+            HostShip.OnAttackFinishAsDefender -= AddBonusAttackAbility;
+        }
+
+        private void PlanBonusAttack()
```

Comment:
Not sure I understand the benefit of doing this in a separate method versus doing it in ActivateAbility?